### PR TITLE
SA surfaced in Secrets table

### PR DESCRIPTION
**Pull request by carlos-logro**
issue: #816 

# Changes
Add Service Account column in Secrets table with respective value per row. Test included.

Screen Shot 2019-12-17 at 3 24 37 PM

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/Secrets/Secrets.js [hunk 1]
```diff
@@ -45,11 +48,11 @@ export /* istanbul ignore next */ class Secrets extends Component {
   constructor(props) {
     super(props);
     this.state = initialState;
-    this.handleCreateSecretClick = this.handleCreateSecretClick.bind(this);
   }
 
   componentDidMount() {
     this.props.fetchSecrets();
+    this.props.fetchServiceAccounts();
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: do we need to re-fetch service accounts when the websocket reconnects in `componentDidUpdate`?

File: src/containers/Secrets/Secrets.js [hunk 2]
```diff
@@ -136,6 +144,13 @@ export /* istanbul ignore next */ class Secrets extends Component {
           defaultMessage: 'Namespace'
         })
       },
+      {
+        key: 'sa',
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: `key: 'serviceAccounts'`

File: src/containers/Secrets/Secrets.js [hunk 3]
```diff
@@ -154,11 +169,20 @@ export /* istanbul ignore next */ class Secrets extends Component {
           }
         });
       }
+      const secretSA = [];
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: naming: let's expand this to `secretServiceAccounts` and similar with other variables below, avoids additional cognitive load interpreting 'sa'/'SA'

File: src/containers/Secrets/Secrets.js [hunk 4]
```diff
@@ -154,11 +169,20 @@ export /* istanbul ignore next */ class Secrets extends Component {
...
+          }
+        });
+      });
       const formattedSecret = {
         annotations,
         id: `${secret.namespace}:${secret.name}`,
         namespace: secret.namespace,
-        name: secret.name
+        name: secret.name,
+        sa: secretSA.join(',')
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: could probably do with a space between too, `.join(', ')`

File: src/containers/Secrets/Secrets.js [hunk 5]
```diff
@@ -279,6 +303,8 @@ function mapStateToProps(state) {
     deleteSuccess: getDeleteSecretsSuccessMessage(state),
     loading: isFetchingSecrets(state),
     secrets: getSecrets(state),
+    fetchingSA: isFetchingServiceAccounts(state),
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: only used with `loading`, could just combine them above and avoid adding an additional prop

`loading: isFetchingSecrets(state) || isFetchingServiceAccounts(state),`

File: src/containers/Secrets/Secrets.js (hunk 1 above)
Comment (carlos-logro) [reply src/containers/Secrets/Secrets.js]: don't really know for sure but ill add it just in case.

**Comment by carlos-logro** [comment]
@AlanGreene thank you for the feedback! all comments have been addressed.

**Comment by AlanGreene** [comment]
Needs a rebase, otherwise should be good to go.

**Comment by carlos-logro** [comment]
@AlanGreene rebased 👍

**Pull request merged**